addons/hr_payroll_account/models/hr_payroll_account.py

```diff
@@ -0,0 +1,171 @@
+#-*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import time
+
+from openerp import api, fields, models, _
+from openerp.tools import float_compare, float_is_zero
+from openerp.exceptions import UserError
+
+
+class HrPayslipLine(models.Model):
+    '''
+    Payslip Line
+    '''
+    _inherit = 'hr.payslip.line'
+
+    @api.model
+    def _get_partner_id(self, credit_account):
+        """
+        Get partner_id of slip line to use in account_move_line
+        """
+        # use partner of salary rule or fallback on employee's address
+        partner_id = self.salary_rule_id.register_id.partner_id.id or self.slip_id.employee_id.address_home_id.id
+        if credit_account:
+            if self.salary_rule_id.register_id.partner_id or self.salary_rule_id.account_credit.type in ('receivable', 'payable'):
+                return partner_id
+        else:
+            if self.salary_rule_id.register_id.partner_id or self.salary_rule_id.account_debit.type in ('receivable', 'payable'):
+                return partner_id
+        return False
+
+
+class HrPayslip(models.Model):
+    '''
+    Pay Slip
+    '''
+    _inherit = 'hr.payslip'
+    _description = 'Pay Slip'
+
+    date_account = fields.Date(string='Date Account', states={'draft': [('readonly', False)]}, readonly=True, help="Keep empty to use the period of the validation(Payslip) date.")
+    journal_id = fields.Many2one('account.journal', string='Salary Journal', states={'draft': [('readonly', False)]}, readonly=True, required=True, default=lambda self: self.env['account.journal'].search([('type', '=', 'general')], limit=1).id)
+    move_id = fields.Many2one('account.move', string='Accounting Entry', readonly=True, copy=False)
+
+    @api.model
+    def create(self, vals):
+        if 'journal_id' in self.env.context:
+            vals.update({'journal_id': self.env.context.get('journal_id')})
+        return super(HrPayslip, self).create(vals)
+
+    @api.multi
+    def onchange_contract_id(self, date_from, date_to, employee_id=False, contract_id=False):
+        HrContract = self.env['hr.contract']
+        res = super(HrPayslip, self).onchange_contract_id(date_from=date_from, date_to=date_to, employee_id=employee_id, contract_id=contract_id)
+        journal_id = self.contract_id.id and HrContract.browse(contract_id).journal_id.id or False
+        res['value'].update({'journal_id': journal_id})
+        return res
+
+    @api.multi
+    def cancel_sheet(self):
+        AccountMove = self.env['account.move']
+        move_ids = []
+        move_to_cancel = []
+        for slip in self.filtered(lambda move: move.move_id):
+            move_ids.append(slip.move_id.id)
+            if slip.move_id.state == 'posted':
+                move_to_cancel.append(slip.move_id.id)
+        AccountMove.button_cancel()
```

Comment:
call button_cancel method with account move  recordset(s) using browse method and pass "move_to_cancel" ..